controllers/workspace/devworkspace_controller.go

```diff
@@ -123,10 +123,13 @@ func (r *DevWorkspaceReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return reconcile.Result{}, err
 	}
 
+	configResolveStatus := &workspaceConditions{}
 	config, err := wkspConfig.ResolveConfigForWorkspace(rawWorkspace, clusterAPI.Client)
 	if err != nil {
-		reqLogger.Error(err, "Error applying external DevWorkspace-Operator configuration")
+		configResolveStatus.setConditionTrue(conditions.DevWorkspaceWarning, fmt.Sprint("Error applying external DevWorkspace-Operator configuration: ", err.Error()))
 		config = wkspConfig.GetGlobalConfig()
+	} else {
+		configResolveStatus.setConditionFalse(conditions.DevWorkspaceWarning, "External DevWorkspaceOperatorConfig successfully resolved")
```

Comment:
I don't know if setting the condition false here is correct -- what this is saying is that there are no warnings (general) for the workspace, not that there's no issue with the config. I think it can be left undefined as it is for other ones (maybe in the future the last thing we do is explicitly set it false at the very end?) -- that way the warning condition doesn't show up at all until there's a warning.